x-pack/elastic-agent/pkg/composable/providers/agent/agent.go

```diff
@@ -27,7 +27,8 @@ func (*contextProvider) Run(comm corecomp.ContextProviderComm) error {
 		return err
 	}
 	err = comm.Set(map[string]interface{}{
-		"id": a.AgentID(),
+		"id":      a.AgentID(),
+		"host_id": a.HostID(),
```

Comment:
I think this should not be added here, and instead it should replace the

`host.id` that is defined in the `host` provider, as this will become the new `host.id` that is consistent.